app/controllers/deploys_controller.rb

```diff
@@ -2,6 +2,10 @@
 require 'csv'
 
 class DeploysController < ApplicationController
+  around_action :wrap_in_with_deleted, if: proc {
```

Comment:
can you turn this into a concern and then move into ApplicationController so we can reuse it anywhere ?